src/execution/vm/bytecode_generator.cpp

```diff
@@ -442,6 +442,55 @@ void BytecodeGenerator::VisitReturnStmt(ast::ReturnStmt *node) {
 void BytecodeGenerator::VisitSqlConversionCall(ast::CallExpr *call, ast::Builtin builtin) {
   ast::Context *ctx = call->GetType()->GetContext();
   switch (builtin) {
+    case ast::Builtin::IsNull: {
+      auto dest = ExecutionResult()->GetOrCreateDestination(ast::BuiltinType::Get(ctx, ast::BuiltinType::Bool));
+      auto input = VisitExpressionForRValue(call->Arguments()[0]);
+      Emitter()->Emit(Bytecode::IsNull, dest, input);
+      ExecutionResult()->SetDestination(dest.ValueOf());
+      break;
+    }
+    case ast::Builtin::IsNotNull: {
+      auto dest = ExecutionResult()->GetOrCreateDestination(ast::BuiltinType::Get(ctx, ast::BuiltinType::Bool));
+      auto input = VisitExpressionForRValue(call->Arguments()[0]);
+      Emitter()->Emit(Bytecode::IsNotNull, dest, input);
+      ExecutionResult()->SetDestination(dest.ValueOf());
+      break;
+    }
+    case ast::Builtin::NullBool: {
```

Comment:
The return types should already be set at this point, so you should just need: ` ExecutionResult()->GetOrCreateDestination(call->GetType()))`. 

Once you do that, these cases are effectively the same and can be macro-d out to make it clearer.

Reply:
Refactored to be one function.